src/ameba/ast/visitors/scope_calls_with_self_receiver_visitor.cr

```diff
@@ -0,0 +1,129 @@
+module Ameba::AST
+  class ScopeCallsWithSelfReceiverVisitor < Crystal::Visitor
```

Comment:
why don't we inherit from BaseVisitor and follow the existing visitor structure?